server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -26,104 +10,112 @@
-        .impressionsMax(getImpressionsMax(row[4]))
-        .isTargetingAge(getAgeTargets(row[5]))
-        .genderTargets(convertStringToList(row[6]))
-        .geoTargets(convertStringToList(row[7]))
-        .spendMin(convertStringToLong(row[8]))
-        .spendMax(convertStringToLong(row[9]))
-        .headline(row[10].trim())
-        .link(row[11].substring(3)) // trim "Ad" from "Ad {URL}"
-        .content(row[12].trim())
-        .headlineSentiment(row[13].trim())
-        .headlineTerms(row[14].trim())
-        .contentSentiment(row[15].trim())
-        .contentTerms(row[16].trim())
+        .id(row[ID_COLUMN])
+        .advertiser(row[ADVERTISER_COLUMN])
+        .startDate(row[START_DATE_COLUMN]) 
+        .endDate(row[END_DATE_COLUMN])
+        .impressionsMin(getImpressionsMin(row[IMPRESSIONS_COLUMN]))
+        .impressionsMax(getImpressionsMax(row[IMPRESSIONS_COLUMN]))
+        .isTargetingAge(getAgeTargets(row[IS_TARGETING_AGE_COLUMN]))
+        .genderTargets(convertStringToList(row[GENDER_TARGETS_COLUMN]))
+        .geoTargets(convertStringToList(row[GEO_TARGETS_COLUMN]))
+        .spendMin(convertStringToLong(row[SPEND_MIN_COLUMN]))
+        .spendMax(convertStringToLong(row[SPEND_MAX_COLUMN]))
+        .headline(row[HEADLINE_COLUMN].trim())
+        .link(row[LINK_COLUMN].substring(3)) // trim "Ad" from "Ad {URL}"
+        .content(row[CONTENT_COLUMN].trim())
+        .headlineSentiment(row[HEADLINE_SENTIMENT_COLUMN].trim())
+        .headlineTerms(row[HEADLINE_TERMS_COLUMN].trim())
+        .contentSentiment(row[CONTENT_SENTIMENT_COLUMN].trim())
+        .contentTerms(row[CONTENT_TERMS_COLUMN].trim())
         .build();
     return ad; 
   }
 
-  /* Min and max impression fields contain letters k and M as well as ≤. The 
-  formatImpressionField helper method makes the fields easier to parse later.
-  */ 
-  public String[] formatImpressionField(String str) {
+  /* 
+   * Min and max impression fields contain letters k and M as well as ≤. The 
+   * formatImpressionField helper method makes the fields easier to parse later.
+   */ 
+  public static String[] formatImpressionField(String str) {
     String impressionsString = str.replace("k", "000").replace("M", "000000").replace(" ","").replace("≤", "");
     String[] impressionsArray = impressionsString.split("-");
     return impressionsArray; 
   }
 
-  public long getImpressionsMin(String impressionsField) throws IllegalArgumentException{
+  public static long getImpressionsMin(String impressionsField) throws IllegalArgumentException {
     String[] impressionsArray = formatImpressionField(impressionsField); 
+    
     // If csv field uses "<=", arr has length 1 and min number of impressions is 0.
     if (impressionsArray.length > 1) {
       try {
         return Long.parseLong(impressionsArray[0]);
-      } catch (IllegalArgumentException e) {
-        return -1; 
+      } catch (NumberFormatException e) {
+        System.out.println("Error: " + e);
```

Comment:
Always throw exceptions, never explicitly exit.  A caller function may want to catch your exceptions and do its own thing with them.  For instance you may write a test to verify that this method does in fact throw the exception, and you'd want to catch the exception and verify that it's correct.  System.exit would prevent that.

You also don't need to System.out.println.  When your program crashes because a thrown exception goes uncaught, it should automatically dump the exception and stack trace.